src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenFunctionPointersTests.cs

```diff
@@ -433,6 +438,928 @@ unsafe struct S
+  IL_0016:  ret
+}");
+        }
+
+        [Fact]
+        public void RefParameter()
+        {
+            var ilStub = @"
+.class public auto ansi beforefieldinit Program
+    extends [mscorlib]System.Object
+{
+    // Methods
+    .method public hidebysig static 
+        method void *(string&) LoadPtr () cil managed 
+    {
+        nop
+        ldftn void Program::Called(string&)
+        ret
+    } // end of method Program::Main
+
+    .method private hidebysig static 
+        void Called (string& arg) cil managed 
+    {
+        nop
+        ldarg.0
+        ldstr ""Ref set""
+        stind.ref
+        ret
+    } // end of Program::Called
+
+    .method public hidebysig specialname rtspecialname
+        instance void .ctor() cil managed
+    {
+            ldarg.0
+            call instance void[mscorlib] System.Object::.ctor()
+            nop
+            ret
+    } // end of Program::.ctor
+}
+";
+
+            var source = @"
+using System;
+unsafe class C
+{
+    static void Main()
+    {
+        delegate*<ref string, void> pointer = Program.LoadPtr();
+        string str = ""Unset"";
+        pointer(ref str);
+        Console.WriteLine(str);
+    }
+}
+";
+
+            var verifier = CompileAndVerifyFunctionPointersWithIl(source, ilStub, expectedOutput: @"Ref set");
+
+            verifier.VerifyIL("C.Main()", expectedIL: @"
+{
+  // Code size       27 (0x1b)
+  .maxstack  2
+  .locals init (delegate*<ref string,void> V_0, //pointer
+                string V_1) //str
+  IL_0000:  call       ""delegate*<ref string,void> Program.LoadPtr()""
+  IL_0005:  stloc.0
+  IL_0006:  ldstr      ""Unset""
+  IL_000b:  stloc.1
+  IL_000c:  ldloca.s   V_1
+  IL_000e:  ldloc.0
+  IL_000f:  calli      0x3
+  IL_0014:  ldloc.1
+  IL_0015:  call       ""void System.Console.WriteLine(string)""
+  IL_001a:  ret
+}");
+        }
+
+        [Fact]
+        public void RefReturnUnused()
+        {
+            var ilStub = @"
+.class public auto ansi beforefieldinit Program
+    extends [mscorlib]System.Object
+{
+    .field public static string 'field'
+
+    // Methods
+    .method public hidebysig static 
+        method string& *() LoadPtr () cil managed 
+    {
+        nop
+        ldftn string& Program::Called()
+        ret
+    } // end of method Program::Main
+
+    .method private hidebysig static 
+        string& Called () cil managed 
+    {
+        nop
+        ldsflda string Program::'field'
+        ret
+    } // end of Program::Called
+
+    .method public hidebysig specialname rtspecialname
+        instance void .ctor() cil managed
+    {
+            ldarg.0
+            call instance void[mscorlib] System.Object::.ctor()
+            nop
+            ret
+    } // end of Program::.ctor
+}
+";
+
+            var source = @"
+using System;
+unsafe class C
+{
+    static void Main()
+    {
+        Program.field = ""Field"";
+        delegate*<ref string> pointer = Program.LoadPtr();
+        Console.WriteLine(pointer());
```

Comment:
>pointer() [](start = 26, length = 9)

How is this unused? Consider renaming the test to RefReturnUseByVal or something similar.